src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,565 @@
+    df3 = pd.DataFrame(in_source_direct_not_in_mondo)
+    if len(df3) == 0:
+        df3 = pd.DataFrame(columns=list(ROBOT_SUBHEADER[0].keys()))
+    # - warning cases
+    #   is.na() checks: only object_mondo_id should ever be missing, but checking both to be safe
+    unmigrated_terms = set(df3[df3['object_mondo_id'].isna()]['object_source_id']).union(
+        set(df3[df3['subject_mondo_id'].isna()]['object_source_id']))
+    if len(unmigrated_terms) >= 200:
+        logging.warning(f'Warning: {len(unmigrated_terms)} unmigrated source terms found in "{ontology_name}"; '
+                        f'higher number than expected.')
+    # - format
+    #   only object_mondo_id should ever be missing, but checking both to be safe
+    df3 = df3[~df3['subject_mondo_id'].isna() & ~df3['object_mondo_id'].isna()].sort_values(common_sort_cols)
+    # - obsolete cases
+    robot_subheader_obs = deepcopy(ROBOT_SUBHEADER)
+    robot_subheader_obs[0]['object_mondo_id'] = 'MONDO:excluded_subClassOf'
+    obsoletes = df3['subject_mondo_label'].str.startswith('obsolete') | df3['object_mondo_label'].str.startswith('obsolete')
+    df3_obs = df3[obsoletes].sort_values(common_sort_cols)
+    # - non-obsolete cases
+    df3 = df3[~obsoletes].sort_values(common_sort_cols)
+    # - save
+    df3_obs = pd.concat([pd.DataFrame(robot_subheader_obs), df3_obs])
+    df3_obs.to_csv(outpath_added_obsolete, sep='\t', index=False)
+    df3 = pd.concat([pd.DataFrame(ROBOT_SUBHEADER), df3])
+    df3.to_csv(outpath_added, sep='\t', index=False)
+
+    # Case 5: SCR is direct in Mondo, and not in the source at all
+    rels_mondo_raw_all: List[RELATIONSHIP] = [x for x in mondo_db.relationships(
+        subjects=list(set(ss_df['subject_id'])), predicates=['rdfs:subClassOf'])]
+    rels_mondo_mondo_all = set(  # filter non-source parents
+        [x for x in rels_mondo_raw_all if any([x[2].startswith(y) for y in MONDO_PREFIX_MAP.keys()])])
+    df5 = pd.DataFrame([{'subject_id': x[0], 'object_id': x[2]} for x in rels_mondo_mondo_all])
+    for pos in ['subject', 'object']:
+        df5[f'{pos}_label'] = df5[f'{pos}_id'].apply(lambda _id: mondo_label_lookup.get(_id, None))
+    src_field = f'in_{source_name}'
+    # unmapped_edges_map: Status of either or both Mondo IDs not mapped to a source. Provides more info than we're
+    #  currently outputting.
+    unmapped_edges_map = {}
+    mapped_edges = set()
+    for e in rels_mondo_mondo_all:
+        key = ''
+        key += 'sub' if e[0] not in mondo_data_map else ''
+        key += 'obj' if e[2] not in mondo_data_map else ''
+        key = 'both' if key == 'subobj' else key
+        if key:
+            unmapped_edges_map.setdefault(key, set()).add(e)
+        else:
+            mapped_edges.add(e)
+    df5[src_field] = df5.apply(
+        # FALSE-MISSING: either or both Mondo IDs not mapped to a source
+        lambda row: 'FALSE-MISSING' if (row['subject_id'], 'rdfs:subClassOf', row['object_id']) not in mapped_edges
```

Comment:
## Case 5 `FALSE-MISSING` @review
Closes:
- #387 

<details><summary>Details</summary>
<p>

I've added; take a look at [google sheet](https://docs.google.com/spreadsheets/d/1eQx1v5_5oE82GULOSeCO_EeN9rC-2tCaKWEWcBXFMG8/edit#gid=728608755).

I've observed something though. All cases of `FALSE-MISSING` are where there is no `object_label`. There's always a `subject_label`.

Now that I think of it, I didn't technically need to add much logic. I could have simply populated `FALSE-MISSING` purely based on presence/absence of the `object_label`.

The reason the subject is never missing is because the source data for this table is based on the set of Mondo IDs in `mondo.sssom.tsv`. If you want, I could change this to the set of all IDs in the Mondo ontology.

</p>
</details>

Reply:
> I've observed something though. All cases of FALSE-MISSING are where there is no object_label. There's always a subject_label.

Mmmmm, this is a bit hacky - but alright, if you think this is robust enough. 

> The reason the subject is never missing is because the source data for this table is based on the set of Mondo IDs in mondo.sssom.tsv. If you want, I could change this to the set of all IDs in the Mondo ontology.

There is something wrong here conceptually. What I was hoping for is this:

Mapping:
MONDO:123 ---> DOID:123
MONDO:124 ----> DOID:124
MONDO:124 ----> ORDO:125

Mondo:
MONDO:123 subclassOf MONDO:124 

Lets say DOID and ORDO are both are entirely empty ontologies.

So when we check MONDO:123 subclassOf MONDO:124, we find it is not in DOID and not in ORDO (since they are both empty).

Now what I care about is this:

In the DOID case, both the subject and the object have mappings. So this is interesting: the terms are both there, and no subclass axiom. Lets call this case: `UNSUPPORTED-SUBCLASS`.

In the ORDO case, the axiom again, is not supported. But for a different reason then in the DOID case: because it is `UNSUPPORTED-MISSING` (either the subject, or the object are not present in ORDO).

These are the two cases I want to distinguish on external source level: `UNSUPPORTED-SUBCLASS` and `UNSUPPORTED-MISSING`. In the aggregate, I will expect this output:

subject | object | case | ordo | doid 
--- | --- | --- | --- | --- 
MONDO:123 | MONDO:124 |  5 | UNSUPPORTED-MISSING | UNSUPPORTED-SUBCLASS

Now I know what I really care about: the subclass is case 5 (entirely unsupported by external source) and DOID _should have had this axiom_ but did not.


For example you have this record in your table:

<google-sheets-html-origin><style type="text/css"><!--td {border: 1px solid #cccccc;}br {mso-data-placement:same-cell;}--></style>
MONDO:0000480 | MONDO:0000477 | True | anismus | focal dystonia | False | False | False | False | False | True |   |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

This is not case 5 as `anismus` is in fact a subClassOf `focal dystonia` in DOID: https://www.ebi.ac.uk/ols4/ontologies/doid/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_0050839 

Can you try to redo the table with the structure above and see if it works? Also, check a few of the records in OLS if they are really case 5.

Reply:
<details><summary>Details</summary>
<p>

Hey @matentzn. Sorry about the miscommunication over our meeting today. It would've been good to talk this over.

I'm going to hold off on working on this because I think it is mostly correct and I get the feeling it's easiest to confirm that at a meeting.

You brought up that I need to do (1) and (2) below, and I brought up that I might want to do (3):
- [x] 1. Unsupported (sub/obj mapped)
- [x] 2. Unsupported (sub &/or obj unmapped)
- [ ] 3. Show all mondo relationships

### 1. Unsupported (sub/obj mapped)

### 2. Unsupported (sub &/or obj unmapped)

### 3. Show all mondo relationships


</p>
</details>

Reply:
I think I know whats wrong in our communication. You think `UNSUPPORTED-SUBCLASS` and `UNSUPPORTED-MISSING` applies on aggregate level (`in_any_source`), while I am talking about source level. I want to know for each source, if they support the axiom because of "MISSING" or "UNSUPPORTED". The aggregate is not so important.

Can you regenerate the table to look like this:

subject_id | object_id | in_any_source | subject_label | object_label | in_icd10cm | in_ordo | in_ncit | in_icd10who | in_omim | in_doid
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
MONDO:0000005 | MONDO:0004907 | False | alopecia, isolated | alopecia | False-missing | False-missing | False-subclass | False-subclass | False-missing | False-missing

Reply:
<details><summary>Details</summary>
<p>

Hmmm... Well, if you look at the table, do you see that the way `in_any_source` is aggregated is based on the values of the `in_SOURCE` columns? The same field values you're asking for are already being populated in these columns as well, and the same mappings apply:

(How it is now --> How you're asking to be changed)
- `True` --> `SUPPORTED`
- `False` --> `UNSUPPORTED-SUBCLASS` || `False - subclass`
- `FALSE-MISSING` --> `UNSUPPORTED-MISSING` || `Falase - missing`

So, the columns do contain the information you're asking for, and I can rename the field values if it will make things clearer.

You didn't suggest it, but perhaps there is some bug a well? I noticed that the only cases of `FALSE-MISSING` in `in_SOURCE` columns show it missing for _all_ sources. This is unexpected, so I'll check this.



</p>
</details>

Reply:
Good news! No miscommunication; just bugs in my logic. I think you'll like [the new output](https://docs.google.com/spreadsheets/d/1eQx1v5_5oE82GULOSeCO_EeN9rC-2tCaKWEWcBXFMG8/edit#gid=1869326589).

I also renamed the field values to those new ones you gave me (e.g. `False --> UNSUPPORTED-SUBCLASS`). Though it's a bit hard to read now. I think I prefer `True`/`False`/`False-Missing`.

Reply:
approved